Mathlib/CategoryTheory/Abelian/SerreClass/MorphismProperty.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Abelian.SerreClass.Basic
+import Mathlib.CategoryTheory.Abelian.DiagramLemmas.KernelCokernelComp
+import Mathlib.CategoryTheory.MorphismProperty.Composition
+import Mathlib.CategoryTheory.MorphismProperty.Retract
+
+/-!
+# The classes of isomorphisms modulo a Serre class
+
+Let `C` be an abelian category and `P : ObjectProperty C` a Serre class.
+We define `P.serreW : MorphismProperty C`, which is the class of
+morphisms `f` such that `kernel f` and `cokernel f` satisfy `P`.
+We show that `P.serreW` is multiplicative, satisfies the two out
+of three property and is stable under retracts.
+
+## TODO
+
+* show that a localized category with respect to `P.serreW` is abelian.
+
+-/
+
+universe v u
+
+namespace CategoryTheory
+
+open Category Limits
+
+variable {C : Type u} [Category.{v} C] [Abelian C]
+
+namespace ObjectProperty
+
+variable (P : ObjectProperty C)
+
+/-- The class of isomorphisms modulo a Serre class: given a
+Serre class `P : ObjectProperty C`, this is the class of morphisms `f`
+such that `kernel f` and `cokernel f` satisfy `P`. -/
+@[nolint unusedArguments]
+def serreW [P.IsSerreClass] : MorphismProperty C :=
```

Comment:
I'm sorry if this is a stupid question, but what does the `W` here (and I guess whenever we use `W` as a name for a morphism property) actually stand for? And how would the name generalize if we ever wanted to generalize to monorphisms/epimorphisms modulo a Serre class?

Reply:
Yes, `serreW` is not a very good name. Maybe, `isoModSerre`?

Reply:
I have just added `monoModSerre` and `epiModSerre`.